### ISSUE TITLE
TypeAhead in floating widgets

### ISSUE THREAD

**Issue by hanbang-wang** [AbdulRahmanAlHamali/flutter_typeahead#88]
The current way of implementing typeahead widget would work correctly for fixed text field, however, for a floating widget like dialogs, the behavior would be incorrect. 

More specifically, when the keyboard popped up, the typeahead would first calculate the height of the typeahead widget, and then the dialog would float to its right position. This would cause the size of the widget to be too small.

My current workaround is to increase the waiting time in `_waitChangeMetrics` so that the dialog would float to the right position and then the height would be calculated correctly.

I don't know if currently there's a better way to address this issue.

**Comment by hanbang-wang** [comment]
A deeper look into the problem shows that it is actually the animation time of dialog being too long (100ms by default), so the default waiting time for TypeAhead is too small.

**Comment by KaYBlitZ** [comment]
Thanks for the bug report. I was able to replicate the problem. Increasing the delay duration does seem to be the proper fix. The only difference is I believe the default time is 150 ms? I took a look at the showDialog function, which is calling showGeneralDialog and passing a transitionDuration of 150 ms.

**Comment by KaYBlitZ** [comment]
Overall I think the fix should be to just change the following lines in `_waitChangeMetrics` to 
```
await Future.delayed(const Duration(milliseconds: 150));
timer += 150;
```

**Comment by hanbang-wang** [comment]
Yes, you are right, it is 150ms. 

So another solution, without changing the source code, is to have a dialog with zero animation time. 

I think the only proper solution for this is to customize an animation with a callback function, and only make typeahead calculate the height when the animation is completed. (thus typeahead needs to provide an interface for that)

Wish someday there will be a global callback like `onAnimationComplete`. Maybe I should suggest this on their issue page.

**Comment by KaYBlitZ** [comment]
That is another solution. I think I will just make the change to 150 in the TypeAhead code, granted that it doesn't break anything else. This way it will just be seamless and work without having the programmer do extra work with the Dialogs.

**Comment by KaYBlitZ** [comment]
Just took a look at #85 . Seems to be the same problem as this. TypeAhead simply breaks whenever there is an animation involved. The height is calculated prematurely before the animation is completed.

You are right. The only real fix is to have some sort of animation completed callback. On our side we can probably expose the resize() call of TypeAhead so that suggestions box can be resized once the animation is completed...

**Comment by KaYBlitZ** [comment]
Can you try my solution to see whether it works? In pubspec add:
```
flutter_typeahead:
    git:
      url: git://github.com/KaYBlitZ/flutter_typeahead.git
      ref: develop
```

**Comment by hanbang-wang** [comment]
It worked (technically). However, 150ms might be too tight and there is some chance of failing. 200ms might be good but there's somehow notably lag.

**Comment by KaYBlitZ** [comment]
I did notice it failing one time. Seems to happen rarely. Only real
solution is if the Google team exposes the animation controller so we can
detect the end or if we can somehow pass our own in. Maybe 170 ms would be
better as a temp fix?

**Comment by sjmcdowall** [comment]
Until a more permanent solution (exposing the animation controller) is available I think we need to live with the 80/20 rule solution for now.. Maybe try 170ms and if that works for 80% of the cases call it "good enough" for now with maybe a "known" issue in the README.md ??

**Comment by KaYBlitZ** [comment]
That seems reasonable. I'll play with it later today and update the PR.
Definitely like the idea of adding it to the README.

**Comment by KaYBlitZ** [comment]
I updated the README and changed the delay to 170 ms. Also noticed something interesting. In the while loop where we are waiting for the keyboard/orientation to change, we are checking whether the change is done before waiting for 170 ms. The small suggestions box might be due to the keyboard toggling too fast and calling resize before we ever reach the delay. I swapped the order so that we are ALWAYS waiting 170 ms FIRST before checking whether the keyboard has really fully toggled or not. I think that will eliminate the problem completely. Performance-wise I didn't really notice a difference with the extra 170 ms wait time.
